packages/rule-utils/src/themeFn.ts

```diff
@@ -0,0 +1,47 @@
+import MagicString from 'magic-string'
+import { colorToString, parseCssColor } from './colors'
+
+export const themeFnRE = /theme\(\s*['"]?(.*?)['"]?\s*\)/g
+
+export function hasThemeFn(str: string) {
+  const result = themeFnRE.test(str)
+  themeFnRE.lastIndex = 0
+  return result
+}
+
+export function transformThemeFn(code: string, theme: Record<string, any>, throwOnMissing = true) {
+  const s = new MagicString(code)
+  const matches = Array.from(code.toString().matchAll(themeFnRE))
+
+  if (!matches.length)
+    return s.toString()
```

Comment:
```suggestion
    return code
```

Reply:
I think we could move `const s = new MagicString(code)` after this condition

Reply:
Good catch!